src/Microsoft.Tye.Extensions/Seq/SeqExtensions.cs

```diff
@@ -0,0 +1,89 @@
+
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Tye.Extensions.Seq
+{
+    internal sealed class SeqExtension : Extension
+    {
+        public override Task ProcessAsync(ExtensionContext context, ExtensionConfiguration config)
+        {
+            if (context.Application.Services.Any(s => s.Name == "seq"))
+            {
+                context.Output.WriteDebugLine("seq service already configured. Skipping...");
+            }
+            else
+            {
+                context.Output.WriteDebugLine("Injecting seq service...");
+
+                var seq = new ContainerServiceBuilder("seq", "datalust/seq")
+                {
+                    EnvironmentVariables =
+                    {
+                        new EnvironmentVariableBuilder("ACCEPT_EULA")
+                        {
+                            Value = "Y"
+                        },
+                    },
+                    Bindings =
+                    {
+                        new BindingBuilder()
+                        {
+                            Name = "seq",
+                            Port = 5340,
+                            ContainerPort = 80,
+                            Protocol = "http",
+                        },
+                    },
+                };
+                context.Application.Services.Add(seq);
+
+                if (config.Data.TryGetValue("logPath", out var obj) &&
+                    obj is string logPath &&
+                    !string.IsNullOrEmpty(logPath))
+                {
+                    seq.Volumes.Add(new VolumeBuilder(logPath, "seq-data", "/data"));
+                }
+
+                foreach (var s in context.Application.Services)
+                {
+                    if (object.ReferenceEquals(s, seq))
+                    {
+                        continue;
+                    }
+
+                    // make seq available as a dependency of everything.
+                    if (!s.Dependencies.Contains(seq.Name))
+                    {
+                        s.Dependencies.Add(seq.Name);
+                    }
+                }
+            }
+
+            if (context.Operation == ExtensionContext.OperationKind.LocalRun)
+            {
+                if (context.Options!.LoggingProvider is null)
+                {
+                    // For local development we hardcode the port and hostname
+                    context.Options.LoggingProvider = "seq=http://localhost:5340";
+                }
+            }
+            else if (context.Operation == ExtensionContext.OperationKind.Deploy)
+            {
+                foreach (var project in context.Application.Services.OfType<DotnetProjectServiceBuilder>())
+                {
+                    var sidecar = DiagnosticAgent.GetOrAddSidecar(project);
+
+                    // Use service discovery to find seq
+                    sidecar.Args.Add("--provider:seq=service:seq");
```

Comment:
Have you tested this?

Reply:
I couldn’t find much information on how this sidecar is implemented. (Maybe the code is in the repository and I totally overlooked it?)

However, I followed the existing patterns from the Elastic extensions and the logs did indeed get pushed into Seq. 

—-

I did test this using the Frontend-backend example and that is how I provided the screenshots. 

I will have a 2nd look to see if I can write up some sort of test to verify more of this functionality

Reply:
This should match up with the provider here: https://github.com/dotnet/tye/blob/master/src/Microsoft.Tye.Hosting.Diagnostics/DiagnosticsProvider.cs#L17

Reply:
Did you deploy and see if it worked?

Reply:
I actually don't have a Kubernetes cluster set up (locally or cloud). Is there any other way to test to see if this works?

Reply:
Not today, we don't support other deployment targets nor do we have CI tests which run with kubernetes. We are working on getting some of that bootstrapped soon.

See https://github.com/dotnet/tye/blob/master/docs/tutorials/hello-tye/01_deploy.md#getting-started-with-deployment for some guidance on how to a cluster setup.

Reply:
The code for the sidecar is here if you want to peep it: https://github.com/dotnet/tye/tree/master/src/tye-diag-agent